src/Chapter3.hs

```diff
@@ -1006,6 +1139,64 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+class Contestant a where
+  fightIsOver :: a -> Bool -- fight is over if the Contestant is either dead or decided to run away. This results in a win for the other contestant
+  handleAttack :: a -> Attack -> a -- the other contestant attacks this contestant with a given Attack
+  applyNextAction :: (Contestant b) => a -> b -> (a, b)
+
+data Knight = Knight {
+  knightHealth :: Health,
+  knightDefense :: Defense,
+  knightActions :: [KnightAction]
+} deriving Show
+
+data KnightAction = AttackContestant Attack | DrinkAHealthPotion Health | CastASpell Defense deriving Show
```

Comment:
Good work on separating `KnighAction` and `MonsterAction` in two separate data types :+1: